### PR TITLE
[full-ci] NATS integration fixes

### PR DESCRIPTION
**Pull request by wkloucek**
## Description
- makes NATS killable by passing a context from the runtime (`ocis kill nats`)
- NATS will die when it encounters an fatal error (like we noticed in the CI during nightlies)

## Related Issue

- Fixes nightly CI issues? Hopefully!

## How Has This Been Tested?

- manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: nats/pkg/server/nats/nats.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	)
+	if err != nil {
+		return err
+	}
+
+	defer n.Shutdown()
+
+	for {
+		// check if NATs server has an encountered an error
+		if err := n.server.LastError(); err != nil {
```
Comment (wkloucek) [inline nats/pkg/server/nats/nats.go:48]:
```
// LastError returns the last fatal error the server experienced.
```

Therefore it should now be killed if an error like in CI occurs:
```
{"level":"fatal","service":"nats","time":"2022-03-04T01:04:43Z","message":"STREAM: Failed to start: read tcp 127.0.0.1:53876->127.0.0.1:9233: i/o timeout"}
```

**Comment by wkloucek** [comment]
@kobergj could you please check if that makes sense? :-) That PR also doesn't pull NATS through REVA anymore, which I find good. Any objections?

File: nats/pkg/server/nats/nats.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+		}
+		// check if the NATs server is still running
+		if n.server.State() == stanServer.Shutdown {
+			return nil
+		}
+		// check if context was cancelled
+		if n.ctx.Err() != nil {
+			return nil
+		}
+		time.Sleep(1 * time.Second)
```
Comment (kobergj) [inline nats/pkg/server/nats/nats.go:59]: Should we make this configurable? Or move it to a global var so we can at least overwrite on build?

File: nats/pkg/server/nats/nats.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+type NATSServer struct {
+	ctx context.Context
+
+	natsOpts *natsServer.Options
+	stanOpts *stanServer.Options
+
+	server *stanServer.StanServer
+}
+
+func NewNATSServer(ctx context.Context, opts ...Option) (*NATSServer, error) {
```
Comment (kobergj) [inline nats/pkg/server/nats/nats.go:20]: Mind adding some comments for the exported funcs in this file?

**Review by kobergj — CHANGES_REQUESTED** [review]
Yes, much better :+1: Just some minors

**Comment by individual-it** [comment]
please run with `full-ci` and best multiple times to prove that it actually helps

**Review by kobergj — APPROVED** [review]

**Pull request merged**